core/controllers/admin_test.py

```diff
@@ -128,12 +128,12 @@ def test_change_rights(self):
         self.signup(
             BOTH_MODERATOR_AND_ADMIN_EMAIL, BOTH_MODERATOR_AND_ADMIN_USERNAME)
 
-        # Navigate to any page. The role is not set.
-        self.testapp.get('/').mustcontain(no=['/moderator', '/admin'])
+        # There should be a redirect on navigating to /
+        self.assertEqual(self.testapp.get('/').status_int, 302)
 
-        # Log in as a superadmin. This gives access to /admin.
+        # Log in as a superadmin. There should again be a redirect on /
         self.login('superadmin@example.com', is_super_admin=True)
-        self.testapp.get('/').mustcontain('/admin', no=['/moderator'])
+        self.assertEqual(self.testapp.get('/').status_int, 302)
```

Comment:
Change this one to

    self.testapp.get('/gallery').mustcontain('/admin', no=['/moderator'])